src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -44,6 +45,12 @@ public TelemetryClient(TelemetryConfiguration configuration)
             }
 
             this.configuration = configuration;
+
+            // If someone created configuration from scratch and forgot to initialize the channel then use the default.
+            if (this.configuration.TelemetryChannel == null)
+            {
+                throw new ArgumentException("The specified configuration does not have a telemetry channel.", "configuration");
```

Comment:
The only way this can throw is if someone explicitly takes steps to create a configuration with the channel set to null (i.e. they make a config instance and then set the channel to null).  No existing code for any customer should be doing that since it was unnecessary previously and makes no sense anyway and all existing code that relied on the default behavior of a channel being created for them will keep working.  Where the channel was created has changed but the fact that it is auto-created has not.  I explicitly made it keep that behavior per Sergey's request.  Throwing now does not break the API, it will not throw for any existing customer or scenario.